tests/ui/numbers-arithmetic/issue-109567.rs

```diff
@@ -0,0 +1,16 @@
+// run-pass
+// check-run-results
+
+pub fn f() -> f64 {
+    std::hint::black_box(-1.0) % std::hint::black_box(-1.0)
+}
+
+pub fn g() -> f64 {
+    -1.0 % -1.0
+}
```

Comment:
To reliably test the interpreter, I'd suggest to put this operation into a `const`.